app/scripts/components/common/datepicker/index.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { ReactNode, useRef, useState, ReactElement } from "react";
+import { DatePicker } from "@trussworks/react-uswds";
+import moment from 'moment';
+
+import './index.scss';
+
+interface SimpleDatePickerProps {
+  disabled: boolean;
+  tipContent: ReactNode;
+  onConfirm: (date: string) => void;
+  triggerHeadReference: string;
+  id: string;
+  selectedDay: Date | null;
+  renderTriggerElement: (props: {
+    onClick: () => void;
+    disabled: boolean;
+    tipContent: ReactNode;
+    triggerHeadReference: string;
+    selectedDay: Date | null;
+  }) => ReactElement;
+}
+
+export const SimpleDatePicker = ({ disabled, tipContent, onConfirm, id, triggerHeadReference, selectedDay, renderTriggerElement }: SimpleDatePickerProps) => {
+    const [showCalendar, setShowCalendar] = useState(true);
+    const datePickerRef = useRef<HTMLDivElement>(null);
+
+    const toggleCalendar = () => {
+      // We have no reference to the internals of the react-uswds DatePicker, so we have to query for it's trigger.
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+      if (datePickerRef.current) {
+        const button = datePickerRef.current.querySelector('button.usa-date-picker__button')! as HTMLButtonElement;
```

Comment:
Ah I see. This is a sticky problem 😬 . Can we control the visibility of the calendar in this way? If we do a conditional render like `{showCalendar && <DatePicker...>}`, we will lose the access to this button. (Calendar is not toggling right now, it just re-renders when renderTriggerElement is pressed I think. )

Reply:
We can't unfortunately, since the `<Calendar />` comes coupled with the `<DatePicker />` and we can't conditionally render only that component. So the only thing I could think of is programmatically toggling it via the `usa-date-picker__button`. 

This gets trickier when we will have to show the `MONTH_PICKER` or the `YEAR_PICKER` based on the dataset time density, because these calendar modes exist as a local state in the Calendar. So the only solution I can think of is to click the picker button to open the calendar, then query and click either `usa-date-picker__calendar__month-selection` or `usa-date-picker__calendar__year-selection` to toggle the corresponding calendar mode depending on the time density 😰 

I opened two issues on `react-uswds` side:

- https://github.com/trussworks/react-uswds/issues/2972
- https://github.com/trussworks/react-uswds/issues/2973

Reply:
Throwing some ideas here. Can it be something that we can solve by overwriting the style? I feel like they (calendar trigger element for both uswds date picker and our date picker) are eventually doing the same thing, showing the date, and then a button to trigger the calendar. (I understand our original design wants the whole thing to be a button to trigger - including the text- but maybe we can adjust the design a bit to accommodate USWDS system?)

ex. overwriting the text input style, calendar icon (to show the caret) here? 

![Screenshot 2024-07-12 at 10 57 42 AM](https://github.com/user-attachments/assets/8a37b588-3952-475d-aba9-0d26065599d9)



![Screenshot 2024-07-12 at 10 57 48 AM](https://github.com/user-attachments/assets/c335563a-e4db-456b-89bc-1062a6f60fc5)

Reply:
Thanks for the idea @hanbyul-here. As you pointed out, we will need to do a lot of manual triggering if we want to sync the calendar with the time density of the dataset, which I think is very error prone. 

> is this a good component to write React components for our own just using uswds style

Yes, I think so. We need a more robust approach but at the same time not to reinvent the wheel (e.g. implementing basic date picker functionality from scratch). So I tried using `react-calendar` (https://www.npmjs.com/package/react-calendar)

Here's a PR with it: https://github.com/NASA-IMPACT/veda-ui/pull/1048 and [here's the deployment](https://deploy-preview-1048--veda-ui.netlify.app/exploration?datasets=%5B%7B%22id%22%3A%22GPM_3IMERGDF.v07%22%2C%22settings%22%3A%7B%22isVisible%22%3Atrue%2C%22opacity%22%3A100%2C%22analysisMetrics%22%3A%5B%7B%22id%22%3A%22mean%22%2C%22label%22%3A%22Average%22%2C%22chartLabel%22%3A%22Average%22%2C%22themeColor%22%3A%22infographicB%22%7D%2C%7B%22id%22%3A%22std%22%2C%22label%22%3A%22St+Deviation%22%2C%22chartLabel%22%3A%22St+Deviation%22%2C%22themeColor%22%3A%22infographicD%22%7D%5D%7D%7D%5D&taxonomy=%7B%7D&search=&date=2024-07-14T22%3A00%3A00.000Z): 

We could override some of the styles using uswds mixins / functions: https://github.com/NASA-IMPACT/veda-ui/pull/1048/files#diff-88b2701907396a34bbe38997d6970f2f6824389dbefe1dcc112a5fdfaa35d811R24

Also:

- It works well with Tippy
- It has out-of-the-box config for showing only the month or year calendar view (it's the `maxDetail`), so it will save us a lot of time

What do you think about this?

Reply:
Yeh, I Agree that we should not invent the wheel. I will vote for using ReactCalendar for now 👍